# 3. Descripción del Conjunto de Datos

## Objetivo
Responder de forma clara y técnica: **¿Qué datos tengo?**

---

## 3.1 Carga de datos

In [1]:
import pandas as pd
import numpy as np
from pathlib import Path

# Cargar dataset limpio
DATA_PATH = Path('../data/processed')
df = pd.read_csv(DATA_PATH / 'nacimientos_2009_2022_limpio.csv')

print("Dataset cargado correctamente")

Dataset cargado correctamente


---
## 3.2 Dimensiones del Dataset

In [3]:
n_observaciones, n_variables = df.shape

print("═" * 50)
print("DIMENSIONES DEL DATASET")
print("═" * 50)
print(f"\n📊 Observaciones (registros): {n_observaciones:,}")
print(f"📋 Variables (columnas):      {n_variables}")
print(f"\n📌 Cada observación representa un NACIMIENTO")
print(f"   ocurrido en Guatemala entre 2009 y 2022.")

══════════════════════════════════════════════════
DIMENSIONES DEL DATASET
══════════════════════════════════════════════════

📊 Observaciones (registros): 5,195,195
📋 Variables (columnas):      36

📌 Cada observación representa un NACIMIENTO
   ocurrido en Guatemala entre 2009 y 2022.


---
## 3.3 Clasificación de Variables

In [4]:
# Clasificación automática por tipo de dato
vars_numericas = df.select_dtypes(include=[np.number]).columns.tolist()
vars_categoricas = df.select_dtypes(include=['object', 'bool']).columns.tolist()

print("═" * 50)
print("CLASIFICACIÓN DE VARIABLES")
print("═" * 50)
print(f"\n📈 Variables numéricas:   {len(vars_numericas)}")
print(f"📝 Variables categóricas: {len(vars_categoricas)}")

══════════════════════════════════════════════════
CLASIFICACIÓN DE VARIABLES
══════════════════════════════════════════════════

📈 Variables numéricas:   30
📝 Variables categóricas: 6


C:\Users\aeh55\AppData\Local\Temp\ipykernel_34444\386324833.py:3: Pandas4Warning: For backward compatibility, 'str' dtypes are included by select_dtypes when 'object' dtype is specified. This behavior is deprecated and will be removed in a future version. Explicitly pass 'str' to `include` to select them, or to `exclude` to remove them and silence this warning.
See https://pandas.pydata.org/docs/user_guide/migration-3-strings.html#string-migration-select-dtypes for details on how to write code that works with pandas 2 and 3.
  vars_categoricas = df.select_dtypes(include=['object', 'bool']).columns.tolist()


### 3.3.1 Variables Numéricas

In [ ]:
# Definición semántica de variables numéricas
descripcion_numericas = {
    # Variables de identificación geográfica (códigos)
    'Depnam': 'Código del departamento de nacimiento de la madre',
    'Depocu': 'Código del departamento donde ocurrió el nacimiento',
    'Depnap': 'Código del departamento de nacimiento del padre',
    'Deprem': 'Código del departamento de residencia de la madre',
    'Deprep': 'Código del departamento de residencia del padre',
    'Depreg': 'Código del departamento de registro',
    'Mupocu': 'Código del municipio donde ocurrió el nacimiento',
    
    # Variables temporales
    'Añoocu': 'Año de ocurrencia del nacimiento',
    'Mesocu': 'Mes de ocurrencia del nacimiento',
    'Diaocu': 'Día de ocurrencia del nacimiento',
    'Añoreg': 'Año de registro del nacimiento',
    'Mesreg': 'Mes de registro del nacimiento',
    
    # Variables demográficas
    'Edadm': 'Edad de la madre (años)',
    'Edadp': 'Edad del padre (años)',
    'Sexo': 'Sexo del recién nacido (1=M, 2=F)',
    
    # Variables del nacimiento
    'Libras': 'Peso al nacer en libras',
    'Onzas': 'Peso al nacer en onzas',
    'Peso_gramos': 'Peso al nacer en gramos (calculado)',
    'Tipar': 'Tipo de parto (1=Simple, 2=Doble, 3=Triple+)',
    
    # Variables de fecundidad
    'Tohite': 'Total de hijos tenidos (incluye el actual)',
    'Tohivi': 'Total de hijos vivos',
    'Tohinm': 'Total de hijos nacidos muertos',
    
    # Variables sociodemográficas (códigos)
    'Escivm': 'Estado civil de la madre (código)',
    'Escivp': 'Estado civil del padre (código)',
    'Asisrec': 'Tipo de asistencia recibida en el parto',
    'Sitioocu': 'Sitio donde ocurrió el nacimiento',
    
    # Variables derivadas (indicadores binarios)
    'Bajo_peso': 'Indicador de bajo peso al nacer (1=Sí)',
    'Madre_adolescente': 'Indicador de madre <20 años (1=Sí)',
    'Madre_35_mas': 'Indicador de madre ≥35 años (1=Sí)',
    'Primer_hijo': 'Indicador de primer hijo (1=Sí)'
}

print("VARIABLES NUMÉRICAS")
print("═" * 70)
print(f"{'Variable':<22} {'Descripción':<48}")
print("-" * 70)

for var in vars_numericas:
    desc = descripcion_numericas.get(var, 'Sin descripción')
    print(f"{var:<22} {desc:<48}")

VARIABLES NUMÉRICAS
══════════════════════════════════════════════════════════════════════
Variable               Descripción                                     
----------------------------------------------------------------------
Diaocu                 Día de ocurrencia del nacimiento                
Depnap                 Código del departamento de nacimiento del padre 
Sexo                   Sexo del recién nacido (1=M, 2=F)               
Depreg                 Código del departamento de registro             
Deprep                 Código del departamento de residencia del padre 
Tohivi                 Total de hijos vivos                            
Depocu                 Código del departamento donde ocurrió el nacimiento
Edadp                  Edad del padre (años)                           
Deprem                 Código del departamento de residencia de la madre
Mesreg                 Mes de registro del nacimiento                  
Mupocu                 Código del municipi

### 3.3.2 Variables Categóricas

In [6]:
# Definición semántica de variables categóricas
descripcion_categoricas = {
    'Grupo_edad_madre': 'Grupo etario de la madre (Adolescente/Adulta joven/Adulta)',
    'Grupo_edad_detallado': 'Rango de edad quinquenal de la madre',
    'Categoria_peso': 'Clasificación del peso según OMS',
    'Sexo_nombre': 'Sexo del recién nacido (texto)',
    'Tipo_parto': 'Tipo de parto (texto)',
    'registro_completo': 'Indica si el registro tiene todas las variables clave'
}

print("VARIABLES CATEGÓRICAS")
print("═" * 70)
print(f"{'Variable':<25} {'Descripción':<45}")
print("-" * 70)

for var in vars_categoricas:
    desc = descripcion_categoricas.get(var, 'Sin descripción')
    print(f"{var:<25} {desc:<45}")

VARIABLES CATEGÓRICAS
══════════════════════════════════════════════════════════════════════
Variable                  Descripción                                  
----------------------------------------------------------------------
registro_completo         Indica si el registro tiene todas las variables clave
Grupo_edad_madre          Grupo etario de la madre (Adolescente/Adulta joven/Adulta)
Grupo_edad_detallado      Rango de edad quinquenal de la madre         
Categoria_peso            Clasificación del peso según OMS             
Sexo_nombre               Sexo del recién nacido (texto)               
Tipo_parto                Tipo de parto (texto)                        


---
## 3.4 Agrupación Temática de Variables

Las variables se organizan en **cinco grupos temáticos** según su naturaleza:

In [ ]:
grupos_tematicos = {
    '1. IDENTIFICACIÓN GEOGRÁFICA': {
        'descripcion': 'Códigos de departamentos y municipios',
        'variables': ['Depreg', 'Depocu', 'Mupocu', 'Deprem', 'Deprep', 'Depnam', 'Depnap'],
        'uso': 'Análisis territorial y espacial'
    },
    '2. TEMPORALES': {
        'descripcion': 'Fechas de ocurrencia y registro',
        'variables': ['Añoocu', 'Mesocu', 'Diaocu', 'Añoreg', 'Mesreg'],
        'uso': 'Análisis de tendencias y estacionalidad'
    },
    '3. CARACTERÍSTICAS DEL NACIMIENTO': {
        'descripcion': 'Datos del recién nacido y el parto',
        'variables': ['Sexo', 'Sexo_nombre', 'Libras', 'Onzas', 'Peso_gramos', 
                      'Categoria_peso', 'Bajo_peso', 'Tipar', 'Tipo_parto', 
                      'Asisrec', 'Sitioocu'],
        'uso': 'Análisis de salud neonatal'
    },
    '4. DATOS DE LA MADRE': {
        'descripcion': 'Características demográficas maternas',
        'variables': ['Edadm', 'Grupo_edad_madre', 'Grupo_edad_detallado',
                      'Madre_adolescente', 'Madre_35_mas', 'Escivm',
                      'Tohite', 'Tohivi', 'Tohinm', 'Primer_hijo'],
        'uso': 'Análisis de patrones de maternidad'
    },
    '5. DATOS DEL PADRE': {
        'descripcion': 'Características demográficas paternas',
        'variables': ['Edadp', 'Escivp'],
        'uso': 'Análisis complementario (alta proporción de faltantes)'
    }
}

print("AGRUPACIÓN TEMÁTICA DE VARIABLES")
print("═" * 70)

for grupo, info in grupos_tematicos.items():
    print(f"\n{grupo}")
    print(f"  📝 {info['descripcion']}")
    print(f"  🎯 Uso: {info['uso']}")
    print(f"  📊 Variables ({len(info['variables'])}): {', '.join(info['variables'])}")

AGRUPACIÓN TEMÁTICA DE VARIABLES
══════════════════════════════════════════════════════════════════════

1. IDENTIFICACIÓN GEOGRÁFICA
  📝 Códigos de departamentos y municipios
  🎯 Uso: Análisis territorial y espacial
  📊 Variables (7): Depreg, Depocu, Mupocu, Deprem, Deprep, Depnam, Depnap

2. TEMPORALES
  📝 Fechas de ocurrencia y registro
  🎯 Uso: Análisis de tendencias y estacionalidad
  📊 Variables (5): Añoocu, Mesocu, Diaocu, Añoreg, Mesreg

3. CARACTERÍSTICAS DEL NACIMIENTO
  📝 Datos del recién nacido y el parto
  🎯 Uso: Análisis de salud neonatal
  📊 Variables (11): Sexo, Sexo_nombre, Libras, Onzas, Peso_gramos, Categoria_peso, Bajo_peso, Tipar, Tipo_parto, Asisrec, Sitioocu

4. DATOS DE LA MADRE
  📝 Características demográficas maternas
  🎯 Uso: Análisis de patrones de maternidad
  📊 Variables (10): Edadm, Grupo_edad_madre, Grupo_edad_detallado, Madre_adolescente, Madre_anosa, Escivm, Tohite, Tohivi, Tohinm, Primer_hijo

5. DATOS DEL PADRE
  📝 Características demográficas patern

---
## 3.5 Resumen Técnico del Dataset

In [8]:
# Calcular estadísticas adicionales
memoria_mb = df.memory_usage(deep=True).sum() / (1024**2)
años_unicos = df['Añoocu'].nunique()
año_min = int(df['Añoocu'].min())
año_max = int(df['Añoocu'].max())

print("""
╔══════════════════════════════════════════════════════════════════════╗
║                    RESUMEN TÉCNICO DEL DATASET                       ║
╠══════════════════════════════════════════════════════════════════════╣""")
print(f"║  📊 Observaciones:        {n_observaciones:>12,} nacimientos              ║")
print(f"║  📋 Variables totales:    {n_variables:>12}                              ║")
print(f"║     • Numéricas:          {len(vars_numericas):>12}                              ║")
print(f"║     • Categóricas:        {len(vars_categoricas):>12}                              ║")
print(f"║  📅 Período:              {año_min} - {año_max} ({años_unicos} años)                   ║")
print(f"║  💾 Memoria en uso:       {memoria_mb:>12.2f} MB                         ║")
print("║                                                                      ║")
print("║  📌 Unidad de análisis: Cada registro = 1 nacimiento                 ║")
print("║  📌 Fuente: Instituto Nacional de Estadística (INE) Guatemala        ║")
print("╚══════════════════════════════════════════════════════════════════════╝")


╔══════════════════════════════════════════════════════════════════════╗
║                    RESUMEN TÉCNICO DEL DATASET                       ║
╠══════════════════════════════════════════════════════════════════════╣
║  📊 Observaciones:           5,195,195 nacimientos              ║
║  📋 Variables totales:              36                              ║
║     • Numéricas:                    30                              ║
║     • Categóricas:                   6                              ║
║  📅 Período:              2009 - 2022 (14 años)                   ║
║  💾 Memoria en uso:            2699.59 MB                         ║
║                                                                      ║
║  📌 Unidad de análisis: Cada registro = 1 nacimiento                 ║
║  📌 Fuente: Instituto Nacional de Estadística (INE) Guatemala        ║
╚══════════════════════════════════════════════════════════════════════╝


---
## 3.6 Tabla Resumen de Variables

Tabla consolidada con información de cada variable:

In [9]:
# Crear tabla resumen
resumen_variables = []

for col in df.columns:
    tipo = 'Numérica' if col in vars_numericas else 'Categórica'
    n_unicos = df[col].nunique()
    n_faltantes = df[col].isna().sum()
    pct_faltantes = (n_faltantes / len(df) * 100)
    
    resumen_variables.append({
        'Variable': col,
        'Tipo': tipo,
        'Valores únicos': n_unicos,
        'Faltantes': n_faltantes,
        '% Faltantes': round(pct_faltantes, 2)
    })

df_resumen = pd.DataFrame(resumen_variables)
print("TABLA RESUMEN DE VARIABLES")
print("═" * 75)
print(df_resumen.to_string(index=False))

TABLA RESUMEN DE VARIABLES
═══════════════════════════════════════════════════════════════════════════
            Variable       Tipo  Valores únicos  Faltantes  % Faltantes
              Diaocu   Numérica              31          0         0.00
              Depnap   Numérica              24          0         0.00
                Sexo   Numérica               2          0         0.00
              Depreg   Numérica              22          0         0.00
              Deprep   Numérica              24          0         0.00
              Tohivi   Numérica              19          0         0.00
              Depocu   Numérica              22          0         0.00
               Edadp   Numérica              82          0         0.00
              Deprem   Numérica              24          0         0.00
              Mesreg   Numérica              12          0         0.00
              Mupocu   Numérica             372          0         0.00
              Tohinm   Numérica  

---

## Síntesis para el Informe

### ¿Qué datos tengo?

El dataset contiene **5,195,195 registros de nacimientos** ocurridos en Guatemala durante el período **2009-2022**, provenientes del Instituto Nacional de Estadística (INE).

**Cada observación representa un nacimiento individual**, con información sobre:

1. **Ubicación geográfica**: Departamento y municipio de ocurrencia y residencia
2. **Temporalidad**: Fecha de ocurrencia y registro
3. **Recién nacido**: Sexo, peso al nacer, tipo de parto
4. **Madre**: Edad, estado civil, historial reproductivo
5. **Padre**: Edad y estado civil (con alta proporción de datos faltantes)

El dataset incluye **36 variables**: 30 numéricas y 6 categóricas, incluyendo variables derivadas creadas durante la limpieza para facilitar el análisis.